Aplib.Core/Intent/Tactics/PrimitiveTactic.cs

```diff
@@ -1,43 +1,44 @@
-﻿using System;
-using Action = Aplib.Core.Intent.Actions.Action;
+﻿using Aplib.Core.Belief;
+using Aplib.Core.Intent.Actions;
 
 namespace Aplib.Core.Intent.Tactics
 {
     /// <summary>
     /// Represents a primitive tactic
     /// </summary>
-    public class PrimitiveTactic : Tactic
+    /// <typeparam name="TBeliefSet">The belief set of the agent.</typeparam>
+    public class PrimitiveTactic<TBeliefSet> : Tactic<TBeliefSet> where TBeliefSet : IBeliefSet
     {
         /// <summary>
         /// Gets the action of the primitive tactic.
         /// </summary>
-        protected readonly Action Action;
+        protected readonly IAction<TBeliefSet> _action;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="PrimitiveTactic"/> class with the specified action.
+        /// Initializes a new instance of the <see cref="PrimitiveTactic{TBeliefSet}"/> class with the specified action.
         /// </summary>
         /// <param name="action">The action of the primitive tactic.</param>
         /// <param name="metadata">
         /// Metadata about this tactic, used to quickly display the tactic in several contexts.
         /// </param>
-        public PrimitiveTactic(Action action, Metadata? metadata = null)
-            : base(metadata) => Action = action;
+        public PrimitiveTactic(IAction<TBeliefSet> action, Metadata? metadata = null)
+            : base(metadata) => _action = action;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="PrimitiveTactic"/> class with the specified action and guard.
+        /// Initializes a new instance of the <see cref="PrimitiveTactic{TBeliefSet}"/> class with the specified action and guard.
         /// </summary>
         /// <param name="action">The action of the primitive tactic.</param>
         /// <param name="guard">The guard of the primitive tactic.</param>
         /// <param name="metadata">
         /// Metadata about this tactic, used to quickly display the tactic in several contexts.
         /// </param>
-        public PrimitiveTactic(Action action, Func<bool> guard, Metadata? metadata = null)
-            : base(guard, metadata) => Action = action;
+        public PrimitiveTactic(IAction<TBeliefSet> action, System.Func<TBeliefSet, bool> guard, Metadata? metadata = null)
```

Comment:
Wordt een beetje een gewoonte maar jullie snappen het denk wel😂. Waarom is `using Action =` weggehaald of Func niet op die manier gedefinieerd?

Reply:
Generic types kun je niet zo definiëren helaas. Erg vaag maar we moeten het ermee doen